main/utils/trophy.c

```diff
@@ -868,26 +867,43 @@ static void _load(trophyDataWrapper_t* tw, const trophyData_t* t)
 
 static void _saveLatestWin(trophyDataWrapper_t* tw)
 {
-    for (int idx = 0; idx < trophySystem.data->length; idx++)
+    int32_t packed = 0;
+    for (int trophyIdx = 0; trophyIdx < trophySystem.data->length; trophyIdx++)
     {
-        if (strcmp(tw->trophyData.title, trophySystem.data->list[idx].title) == 0)
+        if (strcmp(tw->trophyData.title, trophySystem.data->list[trophyIdx].title) == 0)
         {
-            writeNamespaceNvs32(NVSstrings[0], NVSstrings[2], idx);
-            writeNamespaceNvsBlob(NVSstrings[0], NVSstrings[3], trophySystem.data->settings->namespaceKey,
-                                  strlen(trophySystem.data->settings->namespaceKey) + 1);
-            return;
+            for (int i = 0; i < modeListGetCount(); i++)
+            {
+                if (allSwadgeModes[i]->trophyData == NULL)
+                {
+                    continue;
+                }
+                if (strcmp(trophySystem.data->settings->namespaceKey,
+                           allSwadgeModes[i]->trophyData->settings->namespaceKey)
+                    == 0)
+                {
+                    packed = i;
+                    packed = packed << 24;
+                    packed |= trophyIdx;
+                    writeNamespaceNvs32(NVSstrings[0], NVSstrings[2], packed);
+                    return;
+                }
+            }
         }
     }
+    ESP_LOGE("TRO", "Latest trophy did not save!");
 }
 
-int32_t _loadLatestWin(char* buffer)
+static void _loadLatestWin(packedTrophy_t* pt)
 {
-    int32_t idx;
-    size_t blobLen;
-    readNamespaceNvs32(NVSstrings[0], NVSstrings[2], &idx);
-    readNamespaceNvsBlob(NVSstrings[0], NVSstrings[3], NULL, &blobLen);
-    readNamespaceNvsBlob(NVSstrings[0], NVSstrings[3], buffer, &blobLen);
-    return idx;
+    int32_t packed;
+    if (!readNamespaceNvs32(NVSstrings[0], NVSstrings[2], &packed))
+    {
+        pt->trophyIdx = 0x7fffff;
+        return;
+    }
+    pt->trophyIdx = packed & 0x00ffffff;
+    pt->modeIdx   = (packed & 0xff000000) >> 24;
```

Comment:
This was already merged, so nbd, but in the future you can cast `packed` to `packedTrophy_t` (or `&packed` to `packedTrophy_t *`, whichever the compiler is happier with), and read values out that way without bitshifting.